src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -484,6 +492,22 @@ def deployment_options(self):
         nvmes = 0
         res = DeploymentOptions()
         devices = {}
+
+        # setting default values
+        (res.options[OsdDeploymentOptions.COST_CAPACITY.value]
+            .option_name) = 'Cost/Capacity-optimized'
+        (res.options[OsdDeploymentOptions.THROUGHPUT.value]
+            .option_name) = 'Throughput-optimized'
+        (res.options[OsdDeploymentOptions.IOPS.value]
+            .option_name) = 'IOPS-optimized'
+
+        (res.options[OsdDeploymentOptions.COST_CAPACITY.value]
+            .msg) = 'All the available HDDs are selected'
+        (res.options[OsdDeploymentOptions.THROUGHPUT.value]
+            .msg) = 'HDDs/SSDs are selected for data devices and SSDs/NVMes for DB/WAL devices'
+        (res.options[OsdDeploymentOptions.IOPS.value]
+            .msg) = 'All the available NVMes are selected'
```

Comment:
Rather than use procedural programming (assignements, etc.) for initializing static data (all this data is not gonna change, right?), let's lean towards OOP and using the proper classes:
```py
# Named tuples allows us to quickly prototype immutable objects with initialization, defaults and asdict()
class OsdDeploymentOption(NamedTuple):
  id: str
  title: str
  desc: str
  
# Now let's create the enum, where we list and group the different OSD Deployment Options
class OsdDeploymentOptions(Enum):
    COST_CAPACITY = OsdDeploymentOption(
        id='cost-capacity',
        title='Cost/Capacity-optimized',
        desc='All the available HDDs are selected'
    )
        
    THROUGHPUT = OsdDeploymentOption(
        ...
    )

    IOPS = OsdDeploymentOption(
    ...
    )

    # We could even put our "recommender" logic inside a factory method here:
    @classmethod
    def from_host_storage_summaries(cls, host_storage_summaries: List[HostStorageSummary]):
        ret = super_complex_processing_to_guess_the_best_osd_deployment(host_storage_summaries)
        return cls(ret)

>>> OsdDeploymentOptions.from_host_storage_summaries([host_storage_summary_1, ...])
<OsdDeploymentOptions.THROUGHPUT: OsdDeploymentOption(...)>
```

Reply:
Yup, Made the changes (although not used NamedTuple since the values are not constants)

Reply:
We could perhaps introduce `dataclasses` (py3.8 feature) and the corresponding 3.6 [backport package](https://centos.pkgs.org/8/epel-x86_64/python3-dataclasses-0.8-3.el8.noarch.rpm.html)... Those are mutable plain classes with initialization + other helpers.